build/Dockerfile.libbpf.kepler

```diff
@@ -29,7 +29,7 @@ COPY --from=builder /opt/app-root/src/github.com/sustainable-computing-io/kepler
 
 RUN mkdir -p /var/lib/kepler/data
 RUN mkdir -p /var/lib/kepler/bpfassets
-COPY --from=builder /opt/app-root/src/github.com/sustainable-computing-io/kepler/data/normalized_cpu_arch.csv /var/lib/kepler/data/normalized_cpu_arch.csv
+COPY --from=builder /opt/app-root/src/github.com/sustainable-computing-io/kepler/data/cpus.yaml /var/lib/kepler/data/cpus.yaml
```

Comment:
Can we download it from a repository that is actively updated?
I am wondering who will maintain this file updated.

Also, again, can we just use the `/proc/cpuinfo`?

Reply:
> Can we download it from a repository that is actively updated?
> I am wondering who will maintain this file updated.

Yes, this is an issue, however, up till now, I have not found any golang pkg/lib which can do such maintenance.
I once submitted an [issue](https://github.com/klauspost/cpuid/issues/135) in  `github.com/klauspost/cpuid` which is imported in current PR, but the maintainer also could not commit for maintaining the list.

So I preferred to upload the current version of cpus.yaml, and for new coming CPU models, we need to add them into this yaml separately, since nobody could do this, we Kepler maintain them by ourselves...

Due to the public info download feasibility, I had some offline sync with @vimalk78 actually, we found that both the [Intel ark](https://ark.intel.com/content/www/us/en/ark.html#@Processors) and the [WikiChip](https://en.wikichip.org/wiki/WikiChip) were not adequate for the job. 
The former one is Intel only and from HTML, we could only get the mapping of `Brand name` and `uarch`, the `uarch` list itself also comes from our maintained [srcript](https://github.com/sustainable-computing-io/kepler/blob/main/data/download_intel_cpu_models.sh)
The latter one holds more CPU models from different vendors, however the webpage is not friendly for scripts or web crawlers, still huge efforts for such mapping.

So this is why the best choice is leveraging tools such as `cpuid` which is well maintained by [Todd Allen](<todd.allen@etallen.com>) and widely used by Linux Distros. 
Since we are introducing an alternative or fallback plan for `cpuid` here, we can afford such maintenance and I have finished almost the  existed not EOL ones(except some AMD/ARM/RISC-V information which are not easy to find or even cannot support in short-term)

Reply:
> Also, again, can we just use the `/proc/cpuinfo`

Again, the issue for `/proc/cpuinfo` is that there are only five items related to CPU models:
i.e.
vendor_id       : GenuineIntel
cpu family      : 6
model           : 140
model name      : 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
stepping        : 1

Which one could we pick for the `id` of CPU model and without any maintenance efforts? 
There could be various of `model name` for the same microarchitecture, can we afford the efforts for training models for each?

Reply:
@jiere , i think @marceloamaral 's suggestion is to replace `microarchitecture` with a tuple of of `{vendor_id, cpu_family, cpu_model,_no, cpu_model_name, stepping}`. If this tuple maps to a unique `microarchitecture`,  it should be able to serve same purpose as `microarchitecture` in our power model.

Reply:
@vimalk78 , got your point, but actually the information of CPU model finally becomes the metrics labels, if we use tuple, means more labels will be associated to the related metrics, this will bring more confuse to customer and increase later PromQL and kepler-model-server complexity, this is what we should take consideration...

Reply:
hmm, hadn't considered this.

Reply:
@jiere i can get cpu architecture on my laptop via
```console
# cat /sys/devices/cpu/caps/pmu_name
skylake
```

can you confirm if this also works on your end?

Reply:
the kernel code for `pmu_name` is at https://github.com/torvalds/linux/blob/master/arch/x86/events/intel/core.c

Reply:
@rootfs great finding! It works for me. I tested two different systems.
```
$ cat /proc/cpuinfo | grep "model name" -m 1
model name	: Intel(R) Xeon(R) Platinum 8260 CPU @ 2.40GHz
$ cat /sys/devices/cpu/caps/pmu_name
skylake

$ cat /proc/cpuinfo | grep "model name" -m 1
model name	: Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz
$ cat /sys/devices/cpu/caps/pmu_name
haswell
```


Moreover, I was thinking about using the model name `Platinum_8260`,  architecture `skylake`, and the number of chips `2` (used sockets). As in the [SpecPower Data](https://spec.org/power_ssj2008/results/res2023q1/). We might utilize this data later to construct power models, so sticking to this pattern could be beneficial.

Keep in mind that different model names may have different power consumption. But in cases where a specific power model is unavailable, we can choose a model randomly for a given architecture. While this approach may not precisely reflect real power, it can provide to the users some data for reference.

Reply:
That's great! Yes, adding sockets to power models should improve the accuracy. If we do a study of running VMs with different socket configurations (single or multiple), we can get more ideas about the benefits.

Reply:
just observation, there is a difference between value reported by `cpuid` and `pmu_name`

```
vimalkum ~ $ cat /sys/devices/cpu/caps/pmu_name 
icelake
vimalkum ~ $ cpuid -1 | grep uarch
   (uarch synth) = Intel Willow Cove {optim of Sunny Cove}, 10nm++
```

also found this repo https://github.com/archspec/archspec  will try using it and report.
docs: https://archspec.readthedocs.io/en/latest/microarchitectures.html#host-detection

Reply:
Thanks guys for your great findings, I also checked on my various of local machines, several issues found yet:
1. wording issues, just as @vimalk78  found above, the string output is different with `cpuid -1 |grep uarch`
2. For some pretty new CPU models, even though the `cpu model number` has been updated in my current PR, and updated also in kernel [model definition](https://github.com/torvalds/linux/blob/master/arch/x86/include/asm/intel-family.h#L111), but not updated accordingly on `pmu_name` side:
```
# lscpu
Architecture:            x86_64
  CPU op-mode(s):        32-bit, 64-bit
  Address sizes:         45 bits physical, 57 bits virtual
  Byte Order:            Little Endian
CPU(s):                  224
  On-line CPU(s) list:   0-223
Vendor ID:               GenuineIntel
  Model name:            GENUINE INTEL(R) XEON(R)
    CPU family:          6
    Model:               173
    Thread(s) per core:  2
    Core(s) per socket:  56
    Socket(s):           2
    Stepping:            0
```
So the result of `pmu_name` is wrong on that machine:
```
# cat /sys/devices/cpu/caps/pmu_name
sapphire_rapids
```

In a word, all the alternative solutions which leverage the current Linux SYSFS output depends on the related kernel code upstream progress, different part has different implementation, maybe done by different maintainers, the consistence  is a big issue...